sys/net/gnrc/network_layer/ipv6/nib/nib.c

```diff
@@ -1490,6 +1490,62 @@ static inline bool _multihop_p6c(gnrc_netif_t *netif, _nib_abr_entry_t *abr)
 #define _multihop_p6c(netif, abr)   (false)
 #endif  /* CONFIG_GNRC_IPV6_NIB_MULTIHOP_P6C */
 
+static void _configure_subnets(gnrc_netif_t *upstream, const ipv6_addr_t *pfx,
+                               uint8_t pfx_len, uint8_t flags,
+                               uint32_t valid_ltime, uint32_t pref_ltime)
+{
+    if (!IS_USED(MODULE_GNRC_IPV6_NIB_SUBNETS)) {
+        return;
+    }
+
+    unsigned subnets = gnrc_netif_numof() - 1;
+    gnrc_netif_t *downstream = NULL;
+    uint8_t new_pfx_len;
+
+    if (subnets == 0) {
+        return;
+    }
+
+    new_pfx_len = pfx_len + 32 - __builtin_clz(subnets);
+
+    if (new_pfx_len > 64) {
+        DEBUG("nib: can't split /%u into %u subnets\n", pfx_len, subnets);
+        return;
+    }
+
+    while ((downstream = gnrc_netif_iter(downstream))) {
+        ipv6_addr_t new_pfx;
+
+        if (downstream == upstream) {
+            continue;
+        }
```

Comment:
What if there are supposed to be multiple `upstreams`?

Reply:
That's not supported - but it's also not supported by `gnrc_dhcpv6_client_6lbr`.

Reply:
Shouldn't it be possible when using multiple clients, one for each interface?